### PR TITLE
Migrate from CRA to Vite

### PR DESCRIPTION
**Pull request by trevorswanson**
Unfortunately https://github.com/facebook/create-react-app hasn't been updated in almost 2 years, and is now plagued with quite a few security vulnerabilities. Given that PlexSSO is used to secure access to various self-hosted services, I wanted to leverage something that was being actively maintained.

This was admittedly a low-effort transition to [Vite](https://github.com/vitejs/vite), and it can probably be optimized more. However, with CRA/webpack removed, I was able to update all dependencies in the UI to satisfy both Dependabot and `npm audit`.

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -3,18 +3,18 @@
...
     "dependencies": {
+        "@vitejs/plugin-react": "^4.2.1",
         "bootstrap": "^5.3.2",
         "react": "^18.2.0",
-        "react-dom": "^18.2.0"
-    },
-    "devDependencies": {
-        "react-scripts": "^5.0.1"
+        "react-dom": "^18.2.0",
+        "vite": "^5.1.1",
```
Comment (drkno) [inline ui/package.json:10]: nit: 5.1.2?

**Review by drkno — APPROVED** [review]
Nice!

File: ui/package.json (hunk 1 above)
Comment (trevorswanson) [reply ui/package.json:10]: 5.1.2 came out after I submitted my PR, I can test and let you know 😊

**Comment by drkno** [comment]
Fixed in #60

**Pull request closed without merging**